File: backend/app/models/endereco.rb
class Endereco < ApplicationRecord
  belongs_to :cidade
  belongs_to :local, optional: true, touch: true
  belongs_to :fornecedor, optional: true, touch: true
  validates :logradouro, :numero, :bairro, :cep, presence: true

  def self.ransackable_attributes(_auth_object = nil)
    column_names
  end
end

File: frontend/src/app/interfaces/endereco.ts
import { Cidade } from "./cidade";

export interface Endereco {
  id?: number | null,
  logradouro: string,
  numero: string,
  bairro: string,
  complemento: string,
  cep: string,
  cidade_id: number | null,
  principal: boolean,
  created_at?: Date,
  updated_at?: Date,
  cidade?: Cidade
}